lib/internal/fs/streams.js

```diff
@@ -308,16 +308,26 @@ WriteStream.prototype._write = function(data, encoding, cb) {
     });
   }
 
-  fs.write(this.fd, data, 0, data.length, this.pos, (er, bytes) => {
+  const len = data.length;
+  let pos = this.pos;
+  let offset = 0;
+  const callback = (er, bytes) => {
     if (er) {
       if (this.autoClose) {
         this.destroy();
       }
       return cb(er);
     }
     this.bytesWritten += bytes;
-    cb();
-  });
+    pos += bytes;
```

Comment:
Note that `pos` may be `undefined` if `this.pos` was undefined. It would be better to pass `undefined` to `fs.write` rather than `NaN` in this case.